jupyterhub/notebook-images/overlays/cuda-11.0.3/cuda-build-job.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+  name: cuda-11-0-3-build
+spec:
+  backoffLimit: 3
+  template:
+    spec:
+      containers:
+        - image: registry.redhat.io/openshift4/ose-cli:v4.7
+          volumeMounts:
+            - name: cuda-ubi8-build-chain
+              mountPath: /tmp/
+
+          # work around unwriteable HOME dir / for unprivileged pods causing OC commands to be slow in pods
+          env:
+            - name: HOME
+              value: /tmp
+            - name: BUILD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath:
+                    metadata.namespace
+
+          command:
+            - /bin/bash
+            - -c
+            - |
+              set -x
+              echo "PWD: $PWD"
+              oc create -n ${BUILD_NAMESPACE} -f /tmp/gpu-notebook.yaml
```

Comment:
OpenShift recommends using `apply` rather than `create`

Reply:
I am using `oc create` to prevent any situation where this job would re-apply or reset the BuildConfigs and ImageStream when they already exist.

Reply:
Doesn't that prevent us from doing upgrades?

Reply:
A job only runs once (unless it fails), no matter how many times you re-apply it (IIRC), so it does not really matter. But when we want to update the chain, we create a new job and the job would fail as it would not be able to create the resources again.

Reply:
Please open an issue for the re-design or re-factoring of cuda builds and version numbers.